### ISSUE TITLE
[Zoe] Pass in an offerHandle rather than getting it as a return value from `makeInvite`

### ISSUE THREAD

**Issue by DavidBruant** [Agoric/agoric-sdk#840]
All current contracts (i've reviewed them all) have a usage pattern of makeInvites/seats that looks like this: 
```js
const seat = harden({
  method: () => {
    // use offerHandle in some way with reallocate/swap/getOffer/rejectOffer/complete/etc.
  },
});
const { invite, inviteHandle: offerHandle } = zoe.makeInvite(seat);
return invite;
```

`method` has access to `offerHandle` via lexical scoping

If, to make sense, a seat needs to be associated with an offerHandle, maybe that could be reflected by the API:

```js
return zoe.makeInvite(offerHandle => harden({
  method: () => {
    // use offerHandle in some way with reallocate/swap/getOffer/rejectOffer/complete/etc.
  },
});
```

The first argument of `makeInvite` is not the seat anymore, but a `makeSeat(offerHandle)` function

This way, the contract never directly manipulates invite/inviteHandle objects

**Comment by erights** [comment]
The related upcall spike draft is at https://github.com/Agoric/agoric-sdk/pull/700 but needs a lot of work. In a confusing state. Not yet worth looking at. Just fyi for now.

**Comment by katelynsills** [comment]
@erights would this be closed by https://github.com/Agoric/agoric-sdk/pull/853/ when it is a real PR?

**Comment by erights** [comment]
> @erights would this be closed by #853 when it is a real PR?

Yes. Leave it assigned to me.
